packages/dd-trace/src/debugger/devtools_client/snapshot/processor.js

```diff
@@ -0,0 +1,241 @@
+
+  // Perf: Create array of expected size in advance (expect that it contains only one non-enumrable element)
+  const expectedLength = pairs.length - 1
+  const result = { type, entries: new Array(expectedLength) }
+
+  let i = 0
+  for (const pair of pairs) {
+    if (pair.enumerable === false) continue // the value of the `length` property should not be part of the map
+    // The following code is based on assumptions made when researching the output of the Chrome DevTools Protocol.
+    // There doesn't seem to be any documentation to back it up:
+    //
+    // `pair.value` is a special wrapper-object with subtype `internal#entry`. This can be skipped and we can go
+    // directly to its children, of which there will always be exactly two, the first containing the key, and the
+    // second containing the value of this entry of the Map.
+    const key = getPropertyValue(pair.value.properties[0], maxLength)
+    const val = getPropertyValue(pair.value.properties[1], maxLength)
+    result.entries[i++] = [key, val]
+  }
+
+  // Safe-guard in case there were more than one non-enumerable element
+  if (i < expectedLength) result.entries.length = i
+
+  return result
+}
+
+function toSet (type, values, maxLength) {
+  if (values === undefined) return notCaptured(type)
+
+  // Perf: Create array of expected size in advance (expect that it contains only one non-enumrable element)
+  const expectedLength = values.length - 1
+  const result = { type, elements: new Array(expectedLength) }
+
+  let i = 0
+  for (const value of values) {
+    if (value.enumerable === false) continue // the value of the `length` property should not be part of the set
+    // The following code is based on assumptions made when researching the output of the Chrome DevTools Protocol.
+    // There doesn't seem to be any documentation to back it up:
+    //
+    // `value.value` is a special wrapper-object with subtype `internal#entry`. This can be skipped and we can go
+    // directly to its children, of which there will always be exactly one, which contain the actual value in this entry
+    // of the Set.
+    result.elements[i++] = getPropertyValue(value.value.properties[0], maxLength)
+  }
+
+  // Safe-guard in case there were more than one non-enumerable element
+  if (i < expectedLength) result.elements.length = i
+
+  return result
+}
+
+function toArrayBuffer (type, bytes, maxLength) {
+  if (bytes === undefined) return notCaptured(type)
+
+  const size = bytes.length
+
+  if (size > maxLength) {
+    return {
+      type,
+      value: arrayBufferToString(bytes, maxLength),
+      truncated: true,
+      size: bytes.length
+    }
+  } else {
+    return { type, value: arrayBufferToString(bytes, size) }
+  }
+}
+
+function arrayBufferToString (bytes, size) {
+  const buf = Buffer.allocUnsafe(size)
+  for (let i = 0; i < size; i++) {
+    buf[i] = bytes[i].value.value
+  }
+  return buf.toString()
+}
+
+function notCaptured (type) {
```

Comment:
rename to notCaptureDepth (type) or notCapture(type, reason)

Reply:
Fixed